File: macro_tracker/app/schemas/analytics.py
from pydantic import BaseModel
from datetime import date
from typing import List
from .log import Log 

class WeeklySummary(BaseModel):
    start_date: date
    end_date: date
    total_logs: int
    total_calories: float
    avg_daily_calories: float
    avg_daily_protein: float
    avg_daily_carbs: float
    avg_daily_fat: float

    days_calorie_goal_met: int
    avg_calorie_surplus_deficit: float
    avg_protein_surplus_deficit: float

    natural_language_summary: str

class DailySummary(BaseModel):
    date: date
    

    goal_calories: float
    goal_protein: float
    goal_carbs: float
    goal_fat: float
    

    actual_calories: float
    actual_protein: float
    actual_carbs: float
    actual_fat: float
    

    remaining_calories: float
    remaining_protein: float
    remaining_carbs: float
    remaining_fat: float

    logs: List[Log]

File: macro-check-client/src/schemas/analytics.ts
// data str coming from backend
export interface Log { // Add export here
  id: number;
  description: string;
  meal_type: string;
  calories: number;
  date: string;
  protein: number;
  carbs: number;
  fat: number;
}
interface DailyBreakdownItem {
  date: string;
  calories: number;
}
export interface DailySummary {
  date: string;
  goal_calories: number;
  actual_calories: number;
  remaining_calories: number;
  goal_protein: number;
  actual_protein: number;
  remaining_protein: number;
  goal_carbs: number;
  actual_carbs: number;
  remaining_carbs: number;
  goal_fat: number;
  actual_fat: number;
  remaining_fat: number;
  logs: Log[];
}

export interface WeeklySummary {
  start_date: string;
  end_date: string;
  total_logs: number;
  avg_daily_calories: number;
  avg_daily_protein: number;
  days_calorie_goal_met: number;
  avg_calorie_surplus_deficit: number;
  natural_language_summary: string;
  daily_breakdown: DailyBreakdownItem[];
}